src/extensions/surveys.tsx

```diff
@@ -444,6 +445,51 @@ export function usePopupVisibility(
         }
     }, [])
 
+    // Add URL change listener to hide survey when URL no longer matches
+    useEffect(() => {
+        if (isPreviewMode || !survey.conditions?.url || !posthog) {
+            return
+        }
+
+        const checkUrlMatch = () => {
+            const urlMatchType = survey.conditions?.urlMatchType ?? 'icontains'
+            const url = survey.conditions?.url
+            if (!url) {
+                return
+            }
+            const urlCheck = surveyUrlValidationMap[urlMatchType](url)
```

Comment:
not sure if 100% the same but https://github.com/PostHog/posthog-js/blob/4a0fd3142b78dee170274d590b08646039b41c46/src/posthog-surveys.ts#L274-L277
can we extract to a method and reuse in both places?

Reply:
yeah we can, good point, just did with https://github.com/PostHog/posthog-js/pull/1695/commits/31ce8500ca75e5f095b974e1d59c49c8c7553153